src/utils/api.ts

```diff
@@ -33,6 +34,22 @@ export class ApiService {
     this.app.get("/", (_req: Request, res: Response) => {
       res.send("🐮 Moooo!");
     });
+    this.app.get("/health", async (_req: Request, res: Response) => {
```

Comment:
should we expose both the  `/health/liveness` and `/health/readyness` https://www.baeldung.com/linux/kubernetes-livenessprobe-readinessprobe#:~:text=Basics,healthy%20to%20serve%20incoming%20traffic.

Reply:
Sorry, I didn't really make it clear in the PR about my thoughts on the various probes.

The states of the application are:
1. WarmUp
2. Synced

It doesn't _serve_ any data, so there I don't think there is a need for a readiness probe (as there are no ingress' that are being notified to turn on/off request flow to the pod).

Liveness could be exposed, but also if we remove the checks from the watch dog, this makes Kubernetes a requirement (which prevents self-sovereignty). Likely will use a CLI option for `--enable-watchdog` or similar

**NOTE**: A by-product of this is that the liveness probe specification for the IAC repo will require context as to which chain is being configured so it can modify the liveness probes.

Reply:
I think we can just use `/health` for both the startup probe and the liveness probe. No need for a ready probe as it's not being used in this context.

Reply:
What do you mean? In principle both are useful to kubernetes, right? 

so why not to separate and define them? 

What could be the discussion is, if readyness is just for when the API is up, or when the app is in SYNC. Maybe the first is just enough for the test

Reply:
> The kubelet uses readiness probes to know when a container is ready to start accepting traffic. A Pod is considered ready when all of its containers are ready. One use of this signal is to control which Pods are used as backends for Services. When a Pod is not ready, it is removed from Service load balancers.

Readiness is not relevant to the watch-tower as the watch-tower isn't serving data for a kubernetes service, other than the metrics that are being scraped by prometheus. This probe is only useful when you are running pods that are providing services that are being load balanced (like websites, API services etc) and the pod is _temporarily unable to receive / process traffic_ - and signals such to a load balancer. As there is no redundancy running in the replica set, and therefore not using any fancy load balancing (doesn't fit our use case either to start with), there's no need for a discrete `readiness` probe.

This leaves us with `startupProbe` and `livelinessProbe`. The `livelinessProbe` will _not_ run before the `startupProbe` has succeeded. Therefore, as these can't run in parallel, we can actually just use the same API endpoint to provide for these (the configuration of the probes exists in the `Deployment` definition, and there's nothing prohibiting us from using the same endpoint for both).

Reply:
we discussed in person. All good